Lib/email/mime/image.py

```diff
@@ -10,137 +10,143 @@
 from email.mime.nonmultipart import MIMENonMultipart
 
 
+class MIMEImage(MIMENonMultipart):
+    """Class for generating image/* type MIME documents."""
+
+    def __init__(self, _imagedata, _subtype=None,
+                 _encoder=encoders.encode_base64, *, policy=None, **_params):
+        """Create an image/* type MIME document.
+
+        _imagedata is a string containing the raw image data.  If the data
+        type can be detected (jpeg, png, gif, tiff, rgb, pbm, pgm, ppm,
+        rast, xbm, bmp, webp, and exr attempted), then the subtype will be
+        automatically included in the Content-Type header. Otherwise, you can
+        specify the specific image subtype via the _subtype parameter.
+
+        _encoder is a function which will perform the actual encoding for
+        transport of the image data.  It takes one argument, which is this
+        Image instance.  It should use get_payload() and set_payload() to
+        change the payload to the encoded form.  It should also add any
+        Content-Transfer-Encoding or other headers to the message as
+        necessary.  The default encoding is Base64.
+
+        Any additional keyword arguments are passed to the base class
+        constructor, which turns them into parameters on the Content-Type
+        header.
+        """
+        if _subtype is None:
+            if (_subtype := _what(_imagedata)) is None:
+                raise TypeError('Could not guess image MIME subtype')
+        MIMENonMultipart.__init__(self, 'image', _subtype, policy=policy,
+                                  **_params)
+        self.set_payload(_imagedata)
+        _encoder(self)
+
+
+_rules = []
+
+
 # Originally from the imghdr module.
-def _what(h):
-    for tf in tests:
-        if res := tf(h):
+def _what(data):
+    for rule in _rules:
+        if res := rule(data):
             return res
     else:
         return None
 
-tests = []
 
-def _test_jpeg(h):
+def rule(rulefunc):
```

Comment:
```suggestion
def _rule(rulefunc):
```

Do you want to expose this publicly?

Reply:
I thought about that and originally did go with `_rule()` for the decorator name.  I ended up using `rule()` for a couple of reasons:
* `@rule` is more readable
* This leaves the option to advertise `@email.mime.image.rule` for third party extensions in the future (I mean, third parties could do that now but we don't need to document it).
* `__all__` only names `MIMEImage`